index.php

```diff
@@ -10,35 +32,54 @@
 
 
 $PAGE->set_context(context_system::instance());
-$PAGE->set_url('/admin/tool/api_test/index.php');
-$PAGE->set_title(get_string('pluginname', 'tool_api_test'));
+$PAGE->set_url('/admin/tool/wsformat/index.php');
+$PAGE->set_title(get_string('pluginname', 'tool_wsformat'));
 $PAGE->set_heading($SITE->fullname);
 
 echo $OUTPUT->header();
-// echo $OUTPUT->heading(get_string('pluginname', 'tool_api_test'));
-// echo get_string('plugindescription', 'tool_api_test');
-echo $OUTPUT->box_start();
 
-// Prepare the data (context) for the template
-$template_data = new stdClass(); // Create new object
+$output = $PAGE->get_renderer('tool_wsformat');
+
+$plugindescriptiontemplate = new \tool_wsformat\output\plugin_description();
+echo $output->render($plugindescriptiontemplate);
+
+use tool_wsformat\form\autocomplete_form;
+
+$mform = new autocomplete_form();
+$mform->display();
+
+$formarray = [];
 
-// Load the template and render it, passing it $template_data
-// See templates/mytemplate.mustache
-echo $OUTPUT->render_from_template('tool_api_test/mytemplate', $template_data);
+if ($data = $mform->get_data()) {
+    // Populate formarray with selected form web services.
+    foreach ($data->webservice_form as $key => $value) {
+        $formarray[] = (string) $value;
+    }
+}
 
+$selectedsectiontemplate = new \tool_wsformat\output\index_page($formarray);
+$PAGE->requires->js_call_amd('tool_wsformat/test', 'init');
 
-$webservicesObject = $DB->get_records('external_functions', array(), 'name');
-$count = $DB->count_records('external_functions', array());
-echo $count;
-echo '<pre>';
-print_r($webservicesObject);
-echo '</pre>';
+echo $output->render($selectedsectiontemplate);
 
+/**
+ * Function prints webservice function info including parameters and response objects. Used to aid development only.
+ */
+function print_webservices() {
```

Comment:
This is fine to keep existing as a dev tool, but if this is to stick around longer term, this should move into a plugin renderer as a static method